src/bundle/DependencyInjection/Compiler/NavigationHubPass.php

```diff
@@ -24,14 +24,23 @@ public function process(ContainerBuilder $container)
 
     private function processNavigationHubTag(ContainerBuilder $container, $tag, $index)
     {
-        $items = [];
+        $unsorted = [];
         $services = $container->findTaggedServiceIds($tag);
-        foreach (array_keys($services) as $serviceId) {
-            $items[] = new Reference($serviceId);
+        foreach ($services as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
+            }
+            $unsorted[$priority][] = new Reference($serviceId);
+        }
+
+        $sorted = [];
```

Comment:
But `replaceArgument` is a method of Symfony definition. It alters Dependency Injection definition with new argument value.